blog/content/post/2025-05-07-dasharo-ci.md

```diff
@@ -0,0 +1,293 @@
+processor. Microcode updates are critical for firmware security, as they contain
+fixes and mitigations for many classes of exploits, like
+[Spectre and Meltdown](https://meltdownattack.com/).
+
+On the UEFI side, there is UEFI Secure Boot, which is a method for
+cryptographically verifying OS bootloaders before executing them. One of the
+components of UEFI Secure Boot is a revocation database, which is called DBX.
+It contains hashes of revoked binaries, such as bootloaders with known security
+vulnerabilities, like [GRUB and the BootHole exploit](https://eclypsium.com/blog/theres-a-hole-in-the-boot/).
+This mechanism provides a way to revoke binaries that were previously considered
+trusted.
+
+When it comes to overall firmware security trust chain, microcode sits at the
+very beginning, and UEFI Secure Boot sits at the very end, just before the OS
+starts to load. This is why updating these components regularly is crucial for
+platform security, and why we need automatic checks that ensure these components
+are up to date.
+
+In this blog post we'll explore how simple GitHub actions can ensure that these
+critical security components are always up-to-date, helping us deliver a secure
+firmware solution for our users.
+
+## The components
+
+Both Intel Microcode binaries and the UEFI DBX file are publicly available, by
+Intel and the UEFI forum respectively.
+
+Intel Microcode updates are available at Intel's GitHub page:
+[link](https://github.com/intel/Intel-Linux-Processor-Microcode-Data-Files).
+
+From their readme:
+
+> The Intel Processor Microcode Update (MCU) Package provides a mechanism to
+> release updates for security advisories and functional issues, including
+> errata.
+
+The microcode is typically provided by coreboot in a Firmware Interface Table
+(FIT). This table is parsed before the x86 cores begin to execute any
+instructions, which means that the update is performed before coreboot has even
+had a chance to start.
+
+In addition, modern CPUs often depend on ucode updates to _function at all_.
+Due to the amount of erratas, a processor may simply refuse to do anything if
+a microcode update is not provided.
+
+The OS is also able to load microcode updates, but due to how late in the boot
+process that is done, it might be too late to patch some security
+vulnerabilities or erratas.
+
+UEFI DBX is available at UEFI Forum's website: [link](https://uefi.org/revocationlistfile).
+The page describes what the DBX updates are:
+
+> UEFI Revocation List files contain the, now-revoked, signatures of previously
+> approved and signed firmware and software used in booting systems with UEFI
+> Secure Boot enabled.
+
+There is also a GitHub repository hosted by Microsoft, hosting the same files:
+[link](https://github.com/microsoft/secureboot_objects). For convenience, we
+will be using the Microsoft repository, as a git repo is easier to work with
+than a HTML page.
```

Comment:
Ho w can I confirm those are really the same files? I tried hash but apparently I'm doing something wrong.

```
user@work:~/Downloads$ sha256sum DBXUpdate.bin dbxupdate_x64.bin 
37c3d45caa6b061825612215c6dbd06aaacb6f0e426c00bb62b8aee6dd0128de  DBXUpdate.bin
2378fdfe035a8373529ce9acb013fc31b59d3a71d4f9bbbc590bfc8536f90787  dbxupdate_x64.bin
```

Reply:
Hmm I think the UEFI forum version has optional updates applied: https://github.com/microsoft/secureboot_objects/tree/main/PostSignedObjects/Optional/DBX

Although the hashes used to match, clearly something changed recently. Need to investigate

Reply:
I got it, this is the commit that broke it: https://github.com/microsoft/secureboot_objects/commit/ef78acc1b2257bb892655381f8272e6e32d31c3e

sha256sum from before that commit matches that of DBXUpdate from the uefi forum website:

m@phaaze~/Downloads> sha256sum DBXUpdate.bin dbxupdate_x64.bin
2378fdfe035a8373529ce9acb013fc31b59d3a71d4f9bbbc590bfc8536f90787  DBXUpdate.bin
2378fdfe035a8373529ce9acb013fc31b59d3a71d4f9bbbc590bfc8536f90787  dbxupdate_x64.bin

So they updated the github version but neglected to update the UEFI website.

Reply:
@mkopec that is my point, I'm pretty sure MSFT version would be more up to date than UEFI Forum one since MSFT has access to all that data much earlier. We have no idea how process look in background, but we could ask UEFI Forum about that. I would make sure to escalate this topic on X and we will see if there will be anyone who can tell anything.

Reply:
https://github.com/3mdeb/news-and-ideas/pull/659/commits/8787e175de710c0fca0618e32b0fc0b70026fff1

Reply:
I'm resolving this, but there was even recent drama somewhere on HN or X discussing this issue.